spec/models/assignment_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe Assignment, type: :model do
+  describe "associations" do
+    it { should belong_to(:group) }
+    it { should have_many(:projects) }
+  end
+
+  describe "callbacks", :focus do
+    before do
+      @mentor = FactoryBot.create(:user)
+      @group = FactoryBot.create(:group, mentor: @mentor)
+      allow_any_instance_of(Assignment).to receive(:send_new_assignment_mail)
+      allow_any_instance_of(Assignment).to receive(:send_update_mail)
+      allow_any_instance_of(Assignment).to receive(:set_deadline_job)
+    end
+
+    it "should all respective callbacks" do
```

Comment:
You mean "should fire all respective callbacks" ? Anyway this tests seems unnecessary since this is the default rails behavior